contract_comm/evm.go

```diff
@@ -43,31 +41,95 @@ type InternalEVMHandler struct {
 	chain vm.ChainContext
 }
 
-func MakeStaticCall(registryId [32]byte, abi abi.ABI, funcName string, args []interface{}, returnObj interface{}, gas uint64, header *types.Header, state vm.StateDB) (uint64, error) {
-	return makeCallWithContractId(registryId, abi, funcName, args, returnObj, gas, nil, header, state, true)
+func MakeStaticCall(registryId common.Hash, abi abi.ABI, method string, args []interface{}, returnObj interface{}, gas uint64, header *types.Header, state vm.StateDB) (uint64, error) {
+	backend, err := createBackend(header, state)
+	if err != nil {
+		return 0, err
+	}
+
+	return Query(backend, registryId, &abi, method, args, returnObj, gas)
+}
+
+func Query(backend contracts.Backend, registryId common.Hash, abi *abi.ABI, method string, args []interface{}, returnObj interface{}, gas uint64) (uint64, error) {
+
+	contractAddress, err := resolveAddressForCall(backend, registryId, method)
+	if err != nil {
+		return 0, err
+	}
+
+	contract := contracts.NewContract(abi, contractAddress, contracts.SystemCaller)
+	gasLeft, err := contract.Query(contracts.QueryOpts{MaxGas: gas, Backend: backend}, returnObj, method, args...)
```

Comment:
I find the repeated use of the same function name (especially inside itself) a little confusing. It's a common pattern in this PR, but I don't think it gets us much in terms of code re-use (the actual functions) or clarity (which name am I looking at).